### ISSUE TITLE
Classify augment/dataloader

### ISSUE THREAD

**Issue by chdre** [ultralytics/ultralytics#768]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### YOLOv8 Component

Training, Validation

### Bug

Hi!

In the following function. when augment is set to False (test/val), the else part of the following code block will never be used

https://github.com/ultralytics/ultralytics/blob/522f1937edb0728f1fb1ec6398ae56d7f476c3a4/ultralytics/yolo/data/augment.py#L708-L722

due to the following

https://github.com/ultralytics/ultralytics/blob/522f1937edb0728f1fb1ec6398ae56d7f476c3a4/ultralytics/yolo/data/dataset.py#L204

and it will fall back on 

https://github.com/ultralytics/ultralytics/blob/522f1937edb0728f1fb1ec6398ae56d7f476c3a4/ultralytics/yolo/data/augment.py#L683

due to the following

https://github.com/ultralytics/ultralytics/blob/522f1937edb0728f1fb1ec6398ae56d7f476c3a4/ultralytics/yolo/data/dataset.py#L219-L222

Same goes for the yolov5 and respective v5*.py files in this repo. Lastly, if I am not mistaken, shouldn't the image fed into self.torch_transform also be changed to RGB, as it is for self.album_transform?

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@chdre yes that's correct, augmentation is not applied during val/test.

All model inputs should be RGB, and any Albumentations should be in RGB as well.

If you believe there is a bug pleases submit a fix PR with before/after training results on an RGB dataset like Imagenette160 to quantify any improvement or deterioration in performance your proposed changes provide. Thank you!

**Comment by chdre** [comment]
Just posted a PR. One thing is driving me nuts though. The input to torchvision transforms is BGR (non modified opencv image), but the output is RGB. According to the documentation of torchvision transforms it accepts PIL images and arrays, i.e. RGB. I tried modifying the code such that the input in RGB, by the same code that modifies the image going into albumentations, and I get an accuracy of 0.79. I can make a PR for the same code for yolov5 if the PR is accepted, unless you do it yourself.
